cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/BulkWriterContext.java

```diff
@@ -21,7 +21,9 @@
 
 import java.io.Serializable;
 
-public interface BulkWriterContext extends Serializable
+import org.apache.cassandra.spark.common.Reportable;
+
+public interface BulkWriterContext extends Serializable, Reportable
```

Comment:
I don't think this belongs in the `BulkWriterContext`. The context is something we share with executors. However the job stats are tracked from the driver only. I would avoid adding anything else to the BulkWriterContext

Reply:
So, the functionality provided by the new interface is replacing the existing `dialHome` method in the `CassandraBulkWriterContext` .  

The thinking is that this is tied to the "context" that is shared across executors, as we "record" initial stats and job status stats at the executor level and the "inflight" stats at the task level.

Reply:
We could potentially have this implemented by the concrete `CassandraBulkWriterContext` to keep it consistent with how `dialHome` is currently implemented.